### PR TITLE
Add formatting tools for markdown tables

### LINKED ISSUE #299: The tables are stored as HTML when entered via the CMS
Should be something lile:

```
|   | Test | Test |
| --- | --- | --- |
| a | b | c |
| a | b | c |
```

### PR DESCRIPTION
**Pull request by dariomory**
Fixes #299

This issue happens due to a lack of formatting tools (i.e. a header is needed for the table). Added in this PR.

_______

**Changes**

- Added needed formatting tools for markdown tables

### CODE REVIEW

File: src/sdg/conf/base.py [hunk 1]
```diff
@@ -466,6 +464,17 @@
...
 
+# Markdown
+MARKDOWNIFY_WHITELIST_TAGS = [
+    "table",
+    "thead",
+    "tbody",
+    "th",
+    "tr",
+    "td",
+]
```
Comment (joeribekker) [inline src/sdg/conf/base.py:475]: I'm not sure how this works, but we should NOT allow HTML-tags in markdown.

I think we should actually have a validator for those model fields that check for HTML-tags and if present it should give an error that bubbles up in the user interface.

File: src/sdg/js/components/ckeditor.js [hunk 2]
```diff
@@ -26,7 +30,10 @@ ClassicEditor.builtinPlugins = [
     LinkPlugin,
     ListPlugin,
     ParagraphPlugin,
-    Table
+    Table,
+    TableToolbar,
+    TableProperties,
+    TableCellProperties,
```
Comment (joeribekker) [inline src/sdg/js/components/ckeditor.js:36]: We should not allow all sorts of properties.

Markdown only supports tables WITH a header-column but NOT with much else. So merging cells, column-titles, etc. are not allowed.

File: src/sdg/js/components/ckeditor.js [hunk 3]
```diff
@@ -45,5 +52,11 @@ ClassicEditor.defaultConfig = {
             'redo',
         ]
     },
+    table: {
+        contentToolbar: [
+            'tableColumn', 'tableRow', 'mergeTableCells',
+            'tableProperties', 'tableCellProperties'
```
Comment (joeribekker) [inline src/sdg/js/components/ckeditor.js:58]: See comment above.

**Review by joeribekker — CHANGES_REQUESTED** [review]

**Review by joeribekker — CHANGES_REQUESTED** [review]
The issues are not resolved.

**Comment by dariomory** [comment]
> The issues are not resolved.

The current solution does not seem to be ideal. I am looking for a cleaner solution to enforcing the table header, considering the validation approach as a last resort.

File: src/sdg/conf/base.py (hunk 1 above)
Comment (dariomory) [reply src/sdg/conf/base.py:476]: This whitelist is needed to render the markdown tables to html.

**Comment by joeribekker** [comment]
I feel like there's a miscommunication somewhere. I still am allowed to save tables without headers, and add other stuff?

**Review by joeribekker — CHANGES_REQUESTED** [review]
I can still add tables without headers

**Pull request merged**